lib/puppet/util/windows/file.rb

```diff
@@ -397,7 +397,7 @@ def self.resolve_symlink(handle)
   begin
     ffi_lib :kernel32
     attach_function_private :CreateSymbolicLinkW,
-      [:lpwstr, :lpwstr, :dword], :win32_bool
+      [:lpwstr, :lpwstr, :dword], :boolean
```

Comment:
I have compared against the only place where we check the result of the call.  The docs say `returns 0 if failed`, which is what we're checking (though we're using the constant `WIN32_FALSE` which was originally created to compare against `BOOL` and not `BOOLEAN`).

That said, I think we're fine with what we've got.